[AndrewJanong] iP

AndrewJanong opened the pull request:
# Duke
> “Life is too complicated not to be orderly.” – Martha Stewart [(source)](https://dansilvestre.com/productivity-quotes/)

Have a lot of things to do? Fear not as Duke will help you manage your tasks. It's:
* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is:
1. Download
2. Double-click it
3. Add your tasks
4. Let it manage your tasks for you 👍 

Duke is **FREE** to use!

Features:
- [ ] Managing ToDos, Deadlines, and Events 
- [ ] Reminders

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

AndrewJanong committed:
Rename, Greet, Exit

AndrewJanong committed:
Echo feature added

AndrewJanong committed:
Add, List feature added

AndrewJanong committed:
Mark as Done feature added

AndrewJanong committed:
ToDos, Events, Deadlines feature added

AndrewJanong committed:
Automated Text UI Testing

AndrewJanong committed:
Handle Errors by adding Exceptions

AndrewJanong committed:
Delete feature added

AndrewJanong committed:
Used Enums and improved File Management

AndrewJanong committed:
Save feature added

AndrewJanong committed:
Merge branch 'branch-level-7'

AndrewJanong committed:
Dates and Times simple features added

AndrewJanong committed:
Merge branch 'branch-Level-8'

AndrewJanong committed:
Ui class added

AndrewJanong committed:
Added TaskList class and modified Storage class

AndrewJanong committed:
Added Parser class

AndrewJanong committed:
Merge branch 'branch-A-MoreOOP'

AndrewJanong committed:
Put all classes in packages

AndrewJanong committed:
Merge branch 'branch-A-Packages'

AndrewJanong committed:
Merge branch 'add-gradle-support'

AndrewJanong committed:
Added TaskList tests

AndrewJanong committed:
Added Parser tests

AndrewJanong committed:
Merge branch 'branch-A-JUnit'

AndrewJanong committed:
Setup JAR file and fix data.txt creation bug

AndrewJanong committed:
Added JavaDoc comments

AndrewJanong committed:
Follow coding standard

AndrewJanong committed:
Merge branch 'branch-A-JavaDoc'

AndrewJanong committed:
Fix merge conflict

AndrewJanong committed:
Add Find feature

AndrewJanong committed:
Fix merge conflict in Task

TehOPanas commented on src/test/java/duke/parser/ParserTest.java:
Good job on following the test method convention.

TehOPanas commented on src/main/java/duke/Duke.java:
All the codes are organized very well and naming conventions are following the standards. Great job!

AndrewJanong replied:
Thanks!

TehOPanas reviewed (commented):
Follows all the coding standard and have good spacing between codes for high readability.
Naming is also intuitive and straightforward.

AndrewJanong committed:
Improve code style

AndrewJanong committed:
Merge branch 'branch-A-CheckStyle'

AndrewJanong committed:
Create Duke GUI initial setup

limzhenwy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,215 @@
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.ToDo;
+import duke.ui.Ui;
```
for import statements, maybe you could separate the import statements to different chunks.

e.g. one chunk for duke.exceptions, another for duke.task, etc...

would make the import statements look less daunting

AndrewJanong replied:
I actually did this in my previous commits, but checkstyle suggests to not have extra spaces so I just followed 😅

limzhenwy commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,17 @@
+    LIST,
+    MARK,
+    UNMARK,
+
```
could remove this whitespace 
```suggestion
```

limzhenwy reviewed (commented):
LGTM. Just some nits to fix.

AndrewJanong committed:
Finish GUI

AndrewJanong committed:
Remove text-based UI testing

AndrewJanong committed:
Merge branch 'branch-level-10'

AndrewJanong committed:
Add assertions

Without making assertion checks, ensuring the correctness of the assumptions made in the application can be quite challenging.

Assert reassures the assumption made in the program.

To check the command assumption for command exceution functions, assertions can be used to reassure that the command used in each function matches what is needed.

Assertion can also be useful to reassure the index input of task edit functions with the assumption that the input index is a valid index.

Utilizing the Java assert statement serves as an effective method for conducting runtime validations within a program. This enables the early detection of unforeseen issues and failures during the development phase.

AndrewJanong committed:
Improve code quality

AndrewJanong committed:
Merge branch 'branch-A-Assertions'

AndrewJanong committed:
Fix merge conflict

AndrewJanong committed:
Add Help command

AndrewJanong committed:
Merge pull request #2 from AndrewJanong/branch-BCD-Extension

Add Help command

AndrewJanong committed:
Change icon and add background color

AndrewJanong committed:
Add backrgound color to text message

AndrewJanong committed:
Merge branch 'branch-A-BetterGui'

AndrewJanong committed:
Add user guide

AndrewJanong committed:
Fix user guide

AndrewJanong committed:
Prepare jar file for release

AndrewJanong committed:
Edit user guide

Fix quick start and command summary section.

Command Summary table did not render correctly in pages Quick Start list numbers 5 and 6 was replaced by 1 and 2 because of the image in between 1-4 and 5-6.

Fixing these issues will improve user guide quality.

To fix this, some edits have been made. 5 and 6 is replaced by notes and table header was seperated from the table.

AndrewJanong committed:
Fix testing file

AndrewJanong committed:
Fix minor exit command bug
